hcmanager/classification_filter.go

```diff
@@ -0,0 +1,216 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcmanager
+
+import (
+	hwcc "hardware-classification-controller/api/v1alpha1"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+)
+
+// MinMaxFilter it will perform the minimum and maximum comparison based on the value provided by the user and check for the valid host
+func (mgr HardwareClassificationManager) MinMaxFilter(ProfileName string, validatedHost map[string]map[string]interface{}, expectedHardwareprofile hwcc.HardwareCharacteristics) []string {
+	var comparedHost []string
+	for hostname, details := range validatedHost {
+		isHostValid := true
+		for _, value := range details {
+			isValid := false
+			if cpu, CPUOK := value.(bmh.CPU); CPUOK {
+				if checkCPUCount(mgr, cpu, expectedHardwareprofile.Cpu) {
+					isValid = true
+				}
+			}
+			if ram, RAMOK := value.(int64); RAMOK {
+				if checkRAM(mgr, ram, expectedHardwareprofile.Ram) {
+					isValid = true
+				}
+			}
+			if nics, NICSOK := value.(int); NICSOK {
+				if checkNICS(mgr, nics, expectedHardwareprofile.Nic) {
+					isValid = true
+				}
+			}
+			if disks, DISKOK := value.([]bmh.Storage); DISKOK {
+				if checkDiskDetails(mgr, disks, expectedHardwareprofile.Disk) {
+					isValid = true
+				}
+			}
+			if !isValid {
+				isHostValid = false
+				break
+			}
+		}
+		if isHostValid {
+			comparedHost = append(comparedHost, hostname)
```

Comment:
validHosts maybe ? comparedHost is not very explicit and putting it in plural makes clear it is a list

Reply:
True. It should be validHosts